KomaMRIFiles/src/Sequence/Pulseq.jl

```diff
@@ -533,26 +556,42 @@ Reads the RF. It is used internally by [`get_block`](@ref).
 - `shapeLibrary`: (`::Dict{K, V}`) the "shapeLibrary" dictionary
 - `Δt_rf`: (`::Float64`, `[s]`) RF raster time
 - `i`: (`::Int64`) index of the RF in the block event
+- `version_combined`: (`::Int64`) version of the Pulseq file
 
 # Returns
 - `rf`: (`1x1 ::Matrix{RF}`) RF struct
 """
-function read_RF(rfLibrary, shapeLibrary, Δt_rf, i)
+function read_RF(rfLibrary, shapeLibrary, Δt_rf, i, version_combined)
 
     if isempty(rfLibrary) || i==0
         return reshape([RF(0.0,0.0)], 1, 1)
     end
 
     #Unpacking
-    #(1)amplitude (2)mag_id (3)phase_id (4)time_shape_id (5)delay (6)freq (7)phase
     r = rfLibrary[i]["data"]
-    amplitude =     r[1]
-    mag_id =        r[2] |> x->floor(Int64,x)
-    phase_id =      r[3] |> x->floor(Int64,x)
-    time_shape_id = r[4] |> x->floor(Int64,x)
-    delay =         r[5] + (time_shape_id==0)*Δt_rf/2
-    freq =          r[6]
-    phase =         r[7]
+    if version_combined >= 1005000
+        #(1)amplitude (2)mag_id (3)phase_id (4)time_shape_id (5)center (6)delay (7)freq_ppm (8)phase_ppm (9)freq (10)phase (11)use
+        amplitude =     r[1]
+        mag_id =        r[2] |> x->floor(Int64,x)
+        phase_id =      r[3] |> x->floor(Int64,x)
+        time_shape_id = r[4] |> x->floor(Int64,x)
+        center =        r[5]
+        delay =         r[6] + (time_shape_id==0)*Δt_rf/2
+        freq_ppm =      r[7]
+        phase_ppm =     r[8]
+        freq =          r[9]
+        phase =         r[10]
+        use =           r[11]
```

Comment:
You are not using this, there's an issue that could be solved by incorporating this information into the RF.